google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineEnvironment.java

```diff
@@ -23,7 +23,8 @@
  */
 public enum AppEngineEnvironment {
   APP_ENGINE_STANDARD("appengine.environment.name.standard"),
-  APP_ENGINE_FLEX("appengine.environment.name.flexible");
+  APP_ENGINE_FLEX("appengine.environment.name.flexible"),
+  APP_ENGINE_FLEX_COMPAT("appengine.environment.name.flexible");
```

Comment:
Let's say I rename AppEngineEnvironment to AppEngineProjectType. Would that be OK with you guys?

The issue I was mentioning is that we'd have to rename getters/setters everywhere to read e.g., setProjectType instead of setEnvironment. And it could be a tad bit awkward to call `deployConfiguration.setEnvironment(source.getProjectType())`.

That's why I would like to punt this to after the PR.

Regarding making the isFlexCompat checks dynamically, I think there are some places I need the environment where it isn't possible to make them. e.g.,
`AppEngineUtil.setupAppEngineArtifactCombobox`
`AppEngineUtil.createArtifactDeploymentSources`